apiserver/facades/client/applicationoffers/base.go

```diff
@@ -264,16 +258,16 @@ func (api *BaseAPI) checkOfferAccess(ctx context.Context, user coreuser.Name, of
 }
 
 type offerModel struct {
-	model Model
+	model *model.Model
 	err   error
 }
 
 // getModelsFromOffers returns a slice of models corresponding to the
 // specified offer URLs. Each result item has either a model or an error.
-func (api *BaseAPI) getModelsFromOffers(user names.UserTag, offerURLs ...string) ([]offerModel, error) {
+func (api *BaseAPI) getModelsFromOffers(ctx context.Context, user names.UserTag, offerURLs ...string) ([]offerModel, error) {
 	// Cache the models found so far so we don't look them up more than once.
-	modelsCache := make(map[string]Model)
-	oneModel := func(offerURL string) (Model, error) {
+	modelsCache := make(map[string]*model.Model)
+	oneModel := func(offerURL string) (*model.Model, error) {
```

Comment:
If we do have to List the models, we can re-use the List result here.

Reply:
Hi kelvin, sorry do you mind clarifying what do you mean by this?

Reply:
Oh I think you were referring to the model list in modelForName method? For this it has been refactored to getting one model using model name and owner name from model service instead per discussion with Tom. Actually that would have been better, since we would only need to make a single db call.